### PR TITLE
Refactor Risk to Self OASys transformation 

### PR DESCRIPTION
**Pull request by Harriethw**
We would like to check if there is OASys information for an applicant on the Risk to Self section.
If there is data, when the user clicks 'save and continue' we populate the relevant answers in the rest of the task.

[We previously used an existing function ](https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/pull/173)on the CAS API which gets several sections from the report, and does not include one required part of the Risk to Self section (historical risk of self-harm). 

This PR:
* Uses work from this API PR https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/920 to retrieve only the relevant section of OASys
* adds a skeleton 'Vulnerability' page so that we can see answers populating.

Screenshot 2023-08-29 at 14 11 48

Screenshot 2023-08-29 at 14 11 07

### CODE REVIEW

**Comment by Harriethw** [comment]
https://trello.com/c/rT4kUDt2/470-rtos-tech-task-risk-to-self-skeleton

**Comment by Harriethw** [comment]
https://trello.com/c/mzr3lwWs/481-rtos-oasys-guidance-page-mvp

File: server/form-pages/apply/risks-and-needs/risk-to-self/riskToSelfGuidance.ts [hunk 1]
```diff
@@ -68,18 +69,21 @@ export default class RiskToSelfGuidance implements TaskListPage {
...
-    try {
-      oasysSections = await dataServices.personService.getOasysSections(token, application.person.crn)
-
-      taskDataJson = JSON.stringify(RiskToSelfGuidance.getTaskData(oasysSections))
-    } catch (e) {
-      if (e.status === 404) {
-        oasysSections = null
-      } else {
-        throw e
+    if (!application.data['risk-to-self'] || Boolean(Object.keys(application.data['risk-to-self']).length < 1)) {
```
Comment (patrickjfl) [inline server/form-pages/apply/risks-and-needs/risk-to-self/riskToSelfGuidance.ts:76]: just for my own confusion, what case is the second condition covering here?

**Review by patrickjfl — APPROVED** [review]
Nice work 🙂

File: server/form-pages/apply/risks-and-needs/risk-to-self/riskToSelfGuidance.ts (hunk 1 above)
Comment (Harriethw) [reply server/form-pages/apply/risks-and-needs/risk-to-self/riskToSelfGuidance.ts:76]: ah good spot, it doesn't need to be in there, it was just an issue with how integration tests were setup (setting the `risk-to-self` data as empty object, when in real life it wouldn't exist at all)
